System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -75,9 +75,12 @@ public override void Delete(string path)
         public override void Delete(string path, bool recursive)
         {
             path = mockFileDataAccessor.Path.GetFullPath(path).TrimSlashes();
+
+            var pathWithDirectorySeparatorChar = path.Insert(path.Length, Path.DirectorySeparatorChar.ToString());
+
             var affectedPaths = mockFileDataAccessor
                 .AllPaths
-                .Where(p => mockFileDataAccessor.StringOperations.StartsWith(p, path))
+                .Where(p => mockFileDataAccessor.StringOperations.Equals(p,path)|| mockFileDataAccessor.StringOperations.StartsWith(p, pathWithDirectorySeparatorChar))
```

Comment:
Just some missing whitespace:

```suggestion
                 .Where(p => mockFileDataAccessor.StringOperations.Equals(p, path) ||  mockFileDataAccessor.StringOperations.StartsWith(p, pathWithDirectorySeparatorChar))
```

Reply:
e0448ea